configuration/configuration.go

```diff
@@ -202,6 +202,9 @@ type Configuration struct {
 				Deny []string `yaml:"deny,omitempty"`
 			} `yaml:"urls,omitempty"`
 		} `yaml:"manifests,omitempty"`
+		// RepositoryClass validation checks that the class of the configurations inside the
+		// the manifest matches the authorized resource when provided.
+		RepositoryClass bool `yaml:"repositoryclass,omitempty"`
```

Comment:
> This seems like the natural model.

I can make this a list, it ends up having two meanings though, enforce the policy that manifest class must match scope, and enforce manifest class is within this list of classes. When the list is empty do not enforce class at all.